pandas/tests/indexes/test_base.py

```diff
@@ -591,12 +591,19 @@ def test_empty_fancy(self):
             # Index.
             pytest.raises(IndexError, idx.__getitem__, empty_farr)
 
-    def test_getitem(self):
-        arr = np.array(self.dateIndex)
-        exp = self.dateIndex[5]
-        exp = _to_m8(exp)
+    def test_getitem(self, indices):
 
-        assert exp == arr[5]
+        if indices.size != 100:
+            return
+
+        exp = getattr(indices, '_box_func', lambda x: x)(indices._values[0])
```

Comment:
This is a bit too much 'internal-details-relying' for a test, but I didn't directly know a way to write it generic for all types of index instead (or maybe we just shouldn't try, and keep it only testing the errors)

Reply:
you don't need this at all, just check for the error